### ISSUE TITLE
FindToken failure while editing @model directive

### ISSUE THREAD

**Issue by alexgav** [dotnet/roslyn#85511]
I hit this while replacing existing @model directive with a new open via snippet (so the name in model was wrong - default from the snippet)

![image](https://github.com/dotnet/roslyn/assets/1863656/5732d124-e0c4-4687-ac19-e8db16642b54)

I'll see if I can nail down exact repro, but I did capture a dump. Accessible to MSFT employees only.  https://microsoft-my.sharepoint.com/:u:/p/alexgav/EbclQ7M4GrNKiYcwPri7IvMBHNqrCTkO_W4i2Q1LyUYf-Q?e=gi04wI

Basically I replaced existing @model directive via inserting @model snippet, and then hit Tab several times.

![image](https://github.com/dotnet/roslyn/assets/1863656/a71e6fef-2843-4413-bfad-155b0bde8f64)
